spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/iceberg/extensions/IcebergExtendedDataSourceV2Strategy.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.spark.connector.iceberg.extensions;
+
+import com.datastrato.gravitino.spark.connector.iceberg.GravitinoIcebergCatalog;
+import java.util.Collections;
+import org.apache.iceberg.spark.Spark3Util;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalyst.plans.logical.AddPartitionField;
+import org.apache.spark.sql.catalyst.plans.logical.CreateOrReplaceBranch;
+import org.apache.spark.sql.catalyst.plans.logical.CreateOrReplaceTag;
+import org.apache.spark.sql.catalyst.plans.logical.DropBranch;
+import org.apache.spark.sql.catalyst.plans.logical.DropIdentifierFields;
+import org.apache.spark.sql.catalyst.plans.logical.DropPartitionField;
+import org.apache.spark.sql.catalyst.plans.logical.DropTag;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.catalyst.plans.logical.ReplacePartitionField;
+import org.apache.spark.sql.catalyst.plans.logical.SetIdentifierFields;
+import org.apache.spark.sql.catalyst.plans.logical.SetWriteDistributionAndOrdering;
+import org.apache.spark.sql.connector.catalog.CatalogPlugin;
+import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.apache.spark.sql.execution.SparkPlan;
+import org.apache.spark.sql.execution.datasources.v2.AddPartitionFieldExec;
+import org.apache.spark.sql.execution.datasources.v2.CreateOrReplaceBranchExec;
+import org.apache.spark.sql.execution.datasources.v2.CreateOrReplaceTagExec;
+import org.apache.spark.sql.execution.datasources.v2.DropBranchExec;
+import org.apache.spark.sql.execution.datasources.v2.DropIdentifierFieldsExec;
+import org.apache.spark.sql.execution.datasources.v2.DropPartitionFieldExec;
+import org.apache.spark.sql.execution.datasources.v2.DropTagExec;
+import org.apache.spark.sql.execution.datasources.v2.ExtendedDataSourceV2Strategy;
+import org.apache.spark.sql.execution.datasources.v2.ReplacePartitionFieldExec;
+import org.apache.spark.sql.execution.datasources.v2.SetIdentifierFieldsExec;
+import org.apache.spark.sql.execution.datasources.v2.SetWriteDistributionAndOrderingExec;
+import scala.Option;
+import scala.Some;
+import scala.collection.JavaConverters;
+import scala.collection.Seq;
+
+public class IcebergExtendedDataSourceV2Strategy extends ExtendedDataSourceV2Strategy {
+
+  private final SparkSession spark;
+
+  public IcebergExtendedDataSourceV2Strategy(SparkSession spark) {
+    super(spark);
+    this.spark = spark;
+  }
+
+  @Override
+  public Seq<SparkPlan> apply(LogicalPlan plan) {
```

Comment:
spark3.3, 3.4 and 3.5 in Iceberg both support these logical plans, so we don't need to do multiple versions of the implementation.